### DISCUSSION TITLE
Can I create a window that doesn't take the user's focus anyway?

### DISCUSSION THREAD

**Discussion by LeeeSe** [slint-ui/slint#4798] [Q&A]
My App needs to create a window on the Mac that requires the window to not take away the user's focus from the previous application due to a user's click, similar to the Emoji App on the Mac!

<img width="763" alt="image" src="https://github.com/slint-ui/slint/assets/44465325/8e3a9d06-cf08-4203-bb6e-c1ba1b3453b1">

**Comment by LeeeSe** [comment]
Currently I'm using a more cumbersome way to accomplish my needs, but the user experience is not very good

<img width="246" alt="image" src="https://github.com/slint-ui/slint/assets/44465325/1d6111c1-fcfa-4f02-9409-d5e99a123077">

**Reply by bjorn** [reply]
Either typo or clipping issue: "below"

**Reply by LeeeSe** [reply]
Thanks for the heads up.
The reason for this problem is that I set the window width to some fixed value. It was working fine with Chinese fonts, but it looks like it needs some changes!

**Comment by bjorn** [comment]
It's unlikely for Slint to offer a generic API for this, but fortunately you can interact with the backend directly. For example if you're using winit, then see the following issue for a discussion about achieving this: https://github.com/rust-windowing/winit/issues/3072. Also see https://github.com/slint-ui/slint/discussions/4284#discussioncomment-8077128 for a winit and macOS-specific customization example.

**Comment by LeeeSe** [comment]
I apologize for my lack of programming skills.
I've been exploring for hours, but I still haven't figured out how to insert an event_loop into a slint, and this is what I've been able to code so far
```rust
use i_slint_backend_winit::winit::event_loop::EventLoopBuilder;
use i_slint_backend_winit::winit::platform::macos::WindowBuilderExtMacOS;
use i_slint_backend_winit::winit::platform::macos::EventLoopBuilderExtMacOS;

slint::include_modules!();

fn main() -> Result<(), slint::PlatformError> {
    let mut backend = i_slint_backend_winit::Backend::new().unwrap();

    let event_loop = EventLoopBuilder::new()
        .with_activate_ignoring_other_apps(false)
        .build();

    backend.window_builder_hook = Some(Box::new(|builder| {
        builder
            .with_fullsize_content_view(true)
            .with_title_hidden(true)
            .with_titlebar_transparent(true)
    }));


    slint::platform::set_platform(Box::new(backend));
    let ui = AppWindow::new()?;
    let calc = ui.window();

    ui.run()
}
```

**Reply by ogoffart** [reply]
You can use i_slint_backend_winit::[with_event_loop_window_target](https://docs.rs/i-slint-backend-winit/latest/i_slint_backend_winit/fn.with_event_loop_window_target.html#) to access the event loop.  But i'm afraid with_activate_ignoring_other_apps cannot be changed.
We would need to add a hook to access the EventLoopBuilder from the winit backend.

**Reply by LeeeSe** [reply]
I'm currently accomplishing my needs by modifying the event_loop.rs source code directly, it would be great if I could add EventLoopBuilder!

event_loop.rs
```rust
impl NotRunningEventLoop {
    fn new() -> Result<Self, PlatformError> {
        let mut builderr = winit::event_loop::EventLoopBuilder::with_user_event();
        let mut builder = builderr.with_activation_policy(Prohibited);
```

**Reply by bjorn** [reply]
Whoops, sorry I missed that `EventLoopBuilder` was something else than `WindowBuilder`. Good job on solving your problem on the Slint side! For sure it would be nice if this was possible without modifying Slint.

**Comment by LeeeSe** [comment]
If anyone finds that the software cannot be opened after being packaged into an .App 

just add 

```
<key>LSUIElement</key>
<true/>
```

to info.plist
